src/test/java/redis/clients/jedis/tests/commands/AllKindOfValuesCommandsTest.java

```diff
@@ -488,6 +490,25 @@ public void dumpAndRestore() {
     assertTrue(jedis.exists("foo2"));
   }
 
+  @Test(expected = JedisBusyException.class)
+  public void dumpAndRestoreDup() {
+    jedis.set("foo1", "bar1");
+    byte[] sv = jedis.dump("foo1");
+    jedis.restore("foo1", 0, sv);
+    assertTrue(jedis.exists("foo1"));
+  }
+
+  @Test
+  public void dumpAndRestoreReplace() {
```

Comment:
It would be nice to also add `copy` and `keys` tests as well to assert that it also works.